### DISCUSSION TITLE
Can't create an installation access token

### DISCUSSION THREAD

**Discussion by bilelelleuch** [community/community#24525] [Apps, API and Webhooks]
<p>It’s been 2 days trying to create my installation access token with no success.<br>
So as a first step :</p>
<pre><code class="lang-auto">curl -i -H "Authorization: Bearer &lt;JWT&gt;" -H "Accept: application/vnd.github.machine-man-preview+json" https://api.github.com/app
</code></pre>
<p>and I get a</p>
<pre><code class="lang-auto">HTTP/1.1 200 OK
Date: Fri, 12 Jun 2020 09:59:56 GMT
Content-Type: application/json; charset=utf-8
Content-Length: 2492
Server: GitHub.com
Status: 200 OK
...
{
  "id": 68327,
  "slug": "***",
  "node_id": "MDM**=",
  "owner": {
    "login": "bilelelleuch",
    "id": 53218357,

</code></pre>
<p>which means My JWT is OK…<br>
So once I ask for an installation access token via a post request like this :</p>
<pre><code class="lang-auto">curl -i -X POST -H "Authorization: Bearer &lt;JWT&gt;" -H "Accept: application/vnd.github.machine-man-preview+json" https://api.github.com/app/installations/68327/access_tokens
</code></pre>
<p>I receive a</p>
<pre><code class="lang-auto">HTTP/1.1 404 Not Found
Date: Fri, 12 Jun 2020 10:07:09 GMT
Content-Type: application/json; charset=utf-8
Content-Length: 125
Server: GitHub.com
Status: 404 Not Found

</code></pre>
<p>and same for a get request  with <a href="https://api.github.com/app/installations/68327" rel="nofollow noopener">https://api.github.com/app/installations/68327</a></p>

**Comment by francisfuzz** [comment, marked as answer]
<p>:wave: <a class="mention" href="/bilelelleuch">@bilelelleuch</a> , welcome to the GitHub Support Community and for posting this question! :v:</p>
<p>The first request to <code>https://api.github.com/app</code> <a href="https://developer.github.com/v3/apps/#get-the-authenticated-app">returns the authenticated GitHub App</a>, where the value of <code>id</code> of <code>68327</code> represents the identifier for the GitHub App itself and not an installation.</p>
<p>If you’re interested in creating an installation token, you’ll first need to make a request that <a href="https://developer.github.com/v3/apps/#list-installations-for-the-authenticated-app">lists the installations for the authenticated app</a> (you must use a <a href="https://developer.github.com/apps/building-github-apps/authenticating-with-github-apps/#authenticating-as-a-github-app">JWT</a> to access that endpoint). In the <a href="https://developer.github.com/v3/apps/#response-2">response</a>, each element in the collection represents an installation with their own respective unique <code>id</code>s.</p>
<p>Once you’ve determined which <code>id</code> you’re interested in creating an installation token for, you can specify it in the request that previously returned the <code>404 Not Found</code> (replace <code>:installation_id</code>):</p>
<pre><code class="lang-auto">curl -i -X POST -H "Authorization: Bearer &lt;JWT&gt;" -H "Accept: application/vnd.github.machine-man-preview+json" https://api.github.com/app/installations/:installation_id/access_tokens
</code></pre>
<p>Curious to hear how you get on!</p>

**Reply by ikushlianski** [reply]
Thank you, I did not understand that `id` in the `/app` request is not an installation ID but just an ID of the app. Got it working in no time!

**Comment by bilelelleuch** [comment]
<p>Worked perfectly fine.<br>
Thanks a lot ! much appreciated !</p>
